replay/metrics/base_metric.py

```diff
@@ -102,6 +68,66 @@ def preprocess_gt(
     return true_items_by_users
 
 
+def drop_duplicates(recommendations: AnyDataFrame) -> DataFrame:
+
+    """
+    Filter duplicated predictions by choosing the most relevant
+    """
+    return (
+        recommendations.withColumn(
+            "_num",
+            sf.row_number().over(
+                Window.partitionBy("user_idx", "item_idx").orderBy(sf.col("relevance").desc())
+            ),
+        )
+        .where(sf.col("_num") == 1)
+        .drop("_num")
+    )
+
+
+def filter_sort(recommendations: DataFrame, *extra_columns: str) -> DataFrame:
+    """
```

Comment:
could you please explain, why do you use the  `extra_columns` args here, not a single column name? you then define an `extra_col_type` as a type of the first extra column in the list and refer to extra_columns[0] below many times (ignore the rest of the extra columns and return the result for the first column in list only). shall we leave one `extra_column` in params?

Reply:
done